xmtp/src/codecs/text.rs

```diff
@@ -0,0 +1,66 @@
+use super::{CodecError, ContentCodec};
+use std::collections::HashMap;
+use xmtp_proto::xmtp::message_contents::{ContentTypeId, EncodedContent};
+
+pub struct TextCodec {}
+impl TextCodec {
+    const AUTHORITY_ID: &str = "xmtp.org";
+    const TYPE_ID: &str = "text";
+    const ENCODING_KEY: &str = "encoding";
+    const ENCODING_UTF8: &str = "UTF-8";
+}
+
+impl ContentCodec<String> for TextCodec {
+    fn content_type() -> ContentTypeId {
+        ContentTypeId {
+            authority_id: TextCodec::AUTHORITY_ID.to_string(),
```

Comment:
If all content types need to be defined at compile time then it doesn't seem like Strings are needed in this case. Types could be changed to use `&'static str`